### DISCUSSION TITLE
E_IOC_LOOKUP_FAILED:

### DISCUSSION THREAD

**Discussion by mortezabonak** [adonisjs/core#2481] [Help]
I installed an adonis project and want to start to use `node ace` command but got an error
`E_IOC_LOOKUP_FAILED: Cannot resolve "Adonis/Core/BodyParserMiddleware" namespace from the
IoC Container`

package.json:

{
  "name": "store",
  "version": "1.0.0",
  "private": true,
  "scripts": {
    "build": "node ace build --production",
    "start": "node server.js",
    "dev": "node ace serve --watch"
  },
  "devDependencies": {
    "@adonisjs/assembler": "^3.1.1",
    "adonis-preset-ts": "^2.1.0",
    "pino-pretty": "^4.7.1",
    "typescript": "^4.2.4",
    "youch": "^2.2.1",
    "youch-terminal": "^1.1.0"
  },
  "dependencies": {
    "@adonisjs/core": "^5.1.4",
    "@adonisjs/lucid": "^10.0.0",
    "@adonisjs/repl": "^1.1.6",
    "luxon": "^1.26.0",
    "pg": "^8.5.1",
    "proxy-addr": "^2.0.6",
    "reflect-metadata": "^0.1.13",
    "source-map-support": "^0.5.19"
  }
}
I have allready used `@adonisjs/core@5.0.4-preview-rc-2.1` but that's not working too

**Comment by RomainLanz** [comment]
Hey @mortezabonak! 👋 

[I already told you that you were using a not-release version.](https://github.com/adonisjs/core/discussions/2480#discussioncomment-603532)
Please, stick to what the CLI is installing and don't do major version update yourself.

```json
{
  "name": "xxx",
  "version": "1.0.0",
  "private": true,
  "scripts": {
    "build": "node ace build --production",
    "start": "node server.js",
    "dev": "node ace serve --watch"
  },
  "devDependencies": {
    "@adonisjs/assembler": "^3.1.1",
    "adonis-preset-ts": "^2.1.0",
    "pino-pretty": "^4.7.1",
    "typescript": "~4.1",
    "youch": "^2.2.1",
    "youch-terminal": "^1.1.0"
  },
  "dependencies": {
    "@adonisjs/core": "~5.0.4-preview-rc",
    "@adonisjs/repl": "^1.1.6",
    "@adonisjs/session": "^4.0.6",
    "@adonisjs/view": "^3.1.0",
    "proxy-addr": "^2.0.6",
    "reflect-metadata": "^0.1.13",
    "source-map-support": "^0.5.19"
  }
}
```

**Reply by mortezabonak** [reply]
I didn't change the `adonisjs/core` version. this is what it installed.
so I changed to `~5.0.4-preview-rc` but now got new error : 
`TypeError: Cannot read property 'length' of undefined`

**Reply by RomainLanz** [reply]
The versions I shared above are directly from our CLI.
You must have done something to have different version than the one installed.
